### PR TITLE
Make gwt-logger project m2e-compatible

### PR DESCRIPTION
**Pull request by tsmaeder**
This PR restructures the project to work with Eclipse m2e maven tooling. The idea is to move the overridden classes in slf4j to a different source folder and to use the maven-resources-plugin to copy the source files to the output folder (instead of relying on default resource-copy mechanism).
The result is that the maven project can be properly parsed in Eclipse and compilation errors in the overridden classes show up in eclipse.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/5487

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ci-build

File: ide/gwt-logger/pom.xml [hunk 1]
```diff
@@ -42,29 +42,76 @@
...
-                <directory>src/main/resources</directory>
-            </resource>
-            <resource>
-                <directory>${project.build.outputDirectory}/org/eclipse/che/ide/logger/slf4j/emulation</directory>
-            </resource>
-        </resources>
-        <plugins>
-            <plugin>
+	<build>
+		<plugins>
```
Comment (benoitf) [inline ide/gwt-logger/pom.xml]: @tsmaeder you've introduced "tab" characters while the file is formatted with spaces.

what you can do is:
```
$ cd ide/gwt-logger
$ mvn -o com.google.code.sortpom:maven-sortpom-plugin:sort
```

File: ide/gwt-logger/pom.xml [hunk 2]
```diff
@@ -42,29 +42,76 @@
...
+								<resource>
+									<directory>src/main/java</directory>
+									<filtering>false</filtering>
+								</resource>
+							</resources>
+						</configuration>
+					</execution>
+										<execution>
+						<id>copy-emulation</id>
+						<!-- here the phase you need -->
```
Comment (benoitf) [inline ide/gwt-logger/pom.xml]: I think this comment might be removed, it's more like a copy from a tutorial.

**Review by benoitf — CHANGES_REQUESTED** [review]
formatting issue

File: ide/gwt-logger/pom.xml (hunk 2 above)
Comment (tsmaeder) [reply ide/gwt-logger/pom.xml]: yup.

File: ide/gwt-logger/pom.xml (hunk 1 above)
Comment (tsmaeder) [reply ide/gwt-logger/pom.xml]: That's the Eclipse xml formatter set to use tabs.

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
ci-build

**Comment by benoitf** [comment]
ci-build

**Review by skabashnyuk — COMMENTED** [review]
+0 up2 @vparfonov

**Review by vparfonov — APPROVED** [review]

**Pull request merged**